### PR TITLE
Long polling calls

### PR DESCRIPTION
**Pull request by dom-waterson**
long polling done, need to colour in bingo ticket

### CODE REVIEW

File: main-app/app/scripts/bingoController/gameController.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
 (function () {
     'use strict';
     angular.module('Tombola.BingoClient.GameController')
-        .controller('gameController',['proxy', 'userModel', 'bingoModel', 'nextGameService', function (proxy, userModel, bingoModel, nextGameService) {
+        .controller('gameController',['proxy', 'userModel', 'bingoModel', 'nextGameService', 'longPolling', function (proxy, userModel, bingoModel, nextGameService, longPolling) {
```
Comment (keithbarrow) [inline main-app/app/scripts/bingoController/gameController.js]: Not long polling - so don't call it this.

File: main-app/app/scripts/bingoNumbers/callingNumbers.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.LongPollingService')
+        .service('longPolling',['$timeout', 'proxy', 'userModel', 'numberService', function ($timeout, proxy, userModel, numberService) {
```
Comment (keithbarrow) [inline main-app/app/scripts/bingoNumbers/callingNumbers.js]: 1. CamelCase the service names.
2. This doesn't long poll, so don't call it that.

File: main-app/app/scripts/bingoNumbers/callingNumbers.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            var me = this;
+            me.currentNumber = 0;
```
Comment (keithbarrow) [inline main-app/app/scripts/bingoNumbers/callingNumbers.js:9]: Very few of these values are related to polling - can't just whack it all into one service.
- Numbers to win is a responsibility of the ticket/game not the polling infrastructure.

File: main-app/app/scripts/bingoNumbers/callingNumbers.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            me.startNumber = 1;
+            me.winnerMessage ='';
+            me.numbersToWin = 5;
+            me.winnerHasNotBeenFound = true;
+
+            me.startPolling = function() {
+                proxy.getBingoNumber(userModel.token, userModel.name, userModel.currentBalance, me.startNumber).then(function(response) {
+                    me.currentNumber = response.payload.call;
+                    me.startNumber += 1;
+                    checkForWinnerfound(response);
```
Comment (keithbarrow) [inline main-app/app/scripts/bingoNumbers/callingNumbers.js]: Suggestion: have a service to track prizes and prize amounts.

File: main-app/app/scripts/bingoNumbers/callingNumbers.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                    lineFound(data);
+                }
+                else if (data.message === 'Winner') {
+                    fullHouseFound(data);
+                }
+            };
+
+            var fullHouseFound = function (data) {
+                me.winnerHasNotBeenFound = false;
+                me.winnerMessage = "Full house! you have won £" + data.payload.winnerInfo.houseprize;
```
Comment (keithbarrow) [inline main-app/app/scripts/bingoNumbers/callingNumbers.js]: This is embedding UI inside the a service which simplifies communication with the server. This breaks the lasagne-ness of the architecture.

File: main-app/app/scripts/bingoNumbers/callingNumbers.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                userModel.currentBalance += data.payload.winnerInfo.houseprize;
+            };
+
+            var lineFound = function (data) {
+                me.winnerMessage = "Line Found! you have won £" + data.payload.winnerInfo.lineprize;
```
Comment (keithbarrow) [inline main-app/app/scripts/bingoNumbers/callingNumbers.js]: As above - just bind the prize value to wherever you want to display it.

**Comment by dom-waterson** [comment]
did cmd+alt+l to format code so all files changed. Main ones done are callingNumbers.js and checkingWinnerService.js

**Comment by keithbarrow** [comment]
:shipit:

**Pull request merged**